packages/backend-ai/src/conversation/factory.ts

```diff
@@ -29,10 +29,25 @@ class ConversationHandlerFunctionGenerator
   ) {}
 
   generateContainerEntry = ({ scope }: GenerateContainerEntryProps) => {
+    const constructProps: ConversationHandlerFunctionProps = {
+      entry: this.props.entry,
+      models: this.props.models.map((model) => {
+        let modelId;
+        if (typeof model.modelId === 'string') {
+          modelId = model.modelId;
+        } else {
+          modelId = model.modelId.resourcePath;
+        }
```

Comment:
should we validate that this is indeed in the right format and perhaps valid? What if customers provide `{}` or `{ resourcePath: 'rubbish', },`

Reply:
Compiler is going to bark at `{}`. `resourcePath` is mandatory.
The `{ resourcePath: 'rubbish', }` is trickier to guard against, it's typed as `string` because we don't want to constrain it on the client side, given that this list is rather volatile https://docs.aws.amazon.com/bedrock/latest/userguide/model-ids.html .

This additional typing is to make types defined [here](https://github.com/aws-amplify/amplify-api-next/blob/b5b0760cff505e824174ab0b41a00baa735b8744/packages/data-schema/src/ai/AiModelType.ts#L4-L30) flow into this API. But without declaring explicit dependency (just make them equivalent type algebra wise). This is somewhat similar to this solution for function handlers (see [here](https://github.com/aws-amplify/amplify-api-next/blob/b5b0760cff505e824174ab0b41a00baa735b8744/packages/data-schema-types/src/builder/types.ts#L61-L64)).

One possible solution to this might be to lift `AiModel` to `data-schema-types` and use it here. But that's something we should consider for GA, after types on schema mature.